lib/auth/webauthn/login_mfa.go

```diff
@@ -93,29 +92,34 @@ type LoginFlow struct {
 // assertion.
 // As a side effect Begin may assign (and record in storage) a WebAuthn ID for
 // the user.
-func (f *LoginFlow) Begin(ctx context.Context, user string) (*wantypes.CredentialAssertion, error) {
+// Requested challenge extensions will be stored on the stored webauthn challenge
+// record. These extensions indicate additional rules/properties of the webauthn
+// challenge that can be validated in the final login step.
+func (f *LoginFlow) Begin(ctx context.Context, user string, challengeExtensions mfav1.ChallengeExtensions) (*wantypes.CredentialAssertion, error) {
```

Comment:
Maybe:

```suggestion
func (f *LoginFlow) Begin(ctx context.Context, user string, challengeExtensions *mfav1.ChallengeExtensions) (*wantypes.CredentialAssertion, error) {
```

Just because the pointer is the more usual design, plus we can better represent absence if we need to. Same for others.